sources/Debate-frontend/src/app/services/debate.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Observable } from 'rxjs';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { Debate } from '../dto/debate.dto';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DebateService {
+  private endpoint = 'https://localhost:80/api/debate';
+
+  constructor(private http: HttpClient) {}
+
+  getPopularDebates(limit: number, page: number): Observable<Debate[]> {
```

Comment:
What about a private method that accepts 3 parameters:
getDebates(sort: string, limit: number, page: number)
This method would fetch debates from backend. 
DebateService would expose 3 publlic methods (with the same signature as those already present) and invoke the getDebates method with specific sort type.
e.g. getPopularDebates(limit:number, page: number) : Observable<Debate[]>{
return getDebates('popular', limit, page')
}